lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -378,37 +390,75 @@ mod tests {
             .into_generic()
             .expect("convert to generic configuration");
 
-        set_configuration(config);
-
-        set_hostname("agent-test-host".to_string());
+        set_execution_context(ExecutionContext::new(config));
 
         pyo3::append_to_inittab!(datadog_agent);
         pyo3::prepare_freethreaded_python();
 
-        let results = Python::with_gil(|py| -> Result<TestResults, Box<dyn std::error::Error>> {
+        let result = Python::with_gil(|py| -> Result<String, Box<dyn std::error::Error>> {
             let datadog_agent = PyModule::import(py, "datadog_agent")?;
+            let value: String = datadog_agent.getattr("get_config")?.call1(("foo",))?.extract()?;
+            Ok(value)
+        });
+        assert!(result.is_ok(), "Python datadog_agent test failed: {:?}", result.err());
 
-            let hostname: String = datadog_agent.getattr("get_hostname")?.call0()?.extract()?;
+        let value = result.unwrap();
+        assert!(value == "bar", "Config value mismatch: {}", value);
+        std::env::remove_var("DD_TEST_FOO");
+    }
 
-            let config_value: String = datadog_agent.getattr("get_config")?.call1(("foo",))?.extract()?;
+    struct TestResults {
+        hostname: String,
+        tracemalloc_enabled: bool,
+        http_headers: HashMap<String, String>,
+    }
 
-            Ok(TestResults { hostname, config_value })
-        });
+    #[test]
+    fn test_python_checks_config() {
+        trace!("Starting test_python_checks_config");
 
-        assert!(results.is_ok(), "Python datadog_agent test failed: {:?}", results.err());
+        let generic_configuration = ConfigurationLoader::default()
+            .from_yaml("src/sources/checks/builder/python/tests/test_checks_config.yaml")
+            .expect("configuration should be loaded")
+            .into_generic()
+            .expect("convert to generic configuration");
+
+        let execution_context = ExecutionContext::new(generic_configuration).set_hostname("agent-test-host");
+        set_execution_context(execution_context);
+
+        pyo3::append_to_inittab!(datadog_agent);
+        pyo3::prepare_freethreaded_python();
```

Comment:
Drive-by: this seems deprecated: https://docs.rs/pyo3/latest/pyo3/fn.prepare_freethreaded_python.html

Reply:
That is because we are using version 0.25 and the latest version is 0.26